tests/unit/services/test_package.py

```diff
@@ -61,3 +62,168 @@ def test_pack(fake_services: ServiceFactory, default_image_info, mocker):
+            },
+            ["/usr/bin/pebble", "enter"],
+            [],
+        ),
+        # Entrypoint service set
+        (
+            {
+                "run_user": "_daemon_",
+                "environment": {"test": "foo"},
+                "services": {
+                    "test": {"override": "replace", "command": "echo [ foo ]"}
+                },
+                "entrypoint_service": "test",
+            },
+            ["/usr/bin/pebble", "enter", "--args", "test"],
+            ["foo"],
+        ),
+        # Entrypoint command set
+        (
+            {
+                "run_user": "_daemon_",
+                "environment": {"test": "foo"},
+                "services": {"test": {"override": "replace", "command": "echo foo"}},
+                "entrypoint_command": "echo [ foo ]",
+            },
+            ["echo"],
+            ["foo"],
+        ),
+        # Entrypoint command with no CMD
+        (
+            {
+                "run_user": "_daemon_",
+                "environment": {"test": "foo"},
+                "services": {"test": {"override": "replace", "command": "echo foo"}},
+                "entrypoint_command": "echo",
+            },
+            ["echo"],
+            [],
+        ),
+        # Entrypoint command with only CMD
+        (
+            {
+                "run_user": "_daemon_",
+                "environment": {"test": "foo"},
+                "services": {"test": {"override": "replace", "command": "echo foo"}},
+                "entrypoint_command": "[ echo foo ]",
+            },
+            [],
+            ["echo", "foo"],
+        ),
+        # Empty entrypoint command
+        (
+            {
+                "run_user": "_daemon_",
+                "environment": {"test": "foo"},
+                "services": {"test": {"override": "replace", "command": "echo foo"}},
+                "entrypoint_command": "[ ]",
+            },
+            [],
+            [],
+        ),
+        # Entrypoint command edge case
+        (
+            {
+                "run_user": "_daemon_",
+                "environment": {"test": "foo"},
+                "services": {"test": {"override": "replace", "command": "echo foo"}},
+                "entrypoint_command": "echo '[ foo ]' [ bar ]",
+            },
+            ["echo", "[ foo ]"],
+            ["bar"],
+        ),
+    ],
+)
+def test_inner_pack(
+    fake_services: ServiceFactory, mocker, expected_entrypoint, expected_cmd
+):
+    fake_services.get("project").configure(platform=None, build_for=None)
+    project = cast(Project, fake_services.get("project").get())
+
+    # Vars for reuse
+    tag = cast(str, project.version)
+    base_layer_dir = Path()
+    prime_dir = Path("prime")
+    annotations = {"annotation": "foo"}
+    metadata = {"metadata": "bar"}
+
+    # Mock the resulting image and the functions called
+    image = mocker.Mock()
+
+    image.add_user = mocker.Mock()
```

Comment:
I don't think you need to mock all of these individually right? Can't you create a Mock speccing oci.Image?